app/controllers/incomeSources/add/CheckBusinessDetailsController.scala

```diff
@@ -16,62 +16,232 @@
+                                             val authorisedFunctions: AuthorisedFunctions,
+                                             val retrieveNino: NinoPredicate,
+                                             val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                             val itvcErrorHandler: ItvcErrorHandler,
+                                             implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                             val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                             val retrieveBtaNavBar: NavBarPredicate,
+                                               val businessDetailsService: CreateBusinessDetailsService)
+                                            (implicit val ec: ExecutionContext,
+                                             implicit override val mcc: MessagesControllerComponents,
+                                             val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching {
+
+  def show(): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        sources = user.incomeSources,
+        isAgent = false,
+        backUrl = controllers.routes.HomeController.show().url
+      )
+  }
 
   def showAgent(): Action[AnyContent] = Authenticated.async {
     implicit request =>
       implicit user =>
-        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
           implicit mtdItUser =>
             handleRequest(
-              isAgent = true
+              sources = mtdItUser.incomeSources,
+              isAgent = true,
+              backUrl = controllers.routes.HomeController.showAgent.url
             )
         }
   }
 
-}
+  def getDetails(implicit user: MtdItUser[_]): Either[Throwable, CheckBusinessDetailsViewModel] = {
+
+    case class MissingKey(msg: String)
+
+    val errors: Seq[String] = Seq(
+      user.session.data.get("addBusinessName").orElse(Option(MissingKey("MissingKey: addBusinessName"))),
+      user.session.data.get("addBusinessStartDate").orElse(Option(MissingKey("MissingKey: addBusinessStartDate"))),
+      user.session.data.get("addBusinessTrade").orElse(Option(MissingKey("MissingKey: addBusinessTrade"))),
+      user.session.data.get("addBusinessAddressLine1").orElse(Option(MissingKey("MissingKey: addBusinessAddressLine1"))),
+      user.session.data.get("addBusinessPostalCode").orElse(Option(MissingKey("MissingKey: addBusinessPostalCode")))
+    ).collect {
+      case Some(MissingKey(msg)) => MissingKey(msg)
+    }.map(e => e.msg)
+
+    val result: Option[CheckBusinessDetailsViewModel] = for {
+        businessName <- user.session.data.get("addBusinessName")
```

Comment:
we have to use sessionKey defined in the SessionKeys.scala file instead of string values

Reply:
Done